testing/integration/upgrade_test.go

```diff
@@ -162,50 +181,68 @@ func TestStandaloneUpgrade(t *testing.T) {
 		Sudo:    true, // requires Agent installation
 	})
 
-	agentFixture, err := define.NewFixture(t, define.Version())
-
-	require.NoError(t, err)
-
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
-	err = agentFixture.Prepare(ctx)
-	require.NoError(t, err, "error preparing agent fixture")
 
-	err = agentFixture.Configure(ctx, []byte(fastWatcherCfg))
-	require.NoError(t, err, "error configuring agent fixture")
+	versionList := getUpgradableVersions(ctx, t, define.Version())
+
+	for _, parsedVersion := range versionList {
 
-	const minVersionString = "8.9.0-SNAPSHOT"
-	minVersion, _ := version.ParseVersion(minVersionString)
+		if parsedVersion.String() == define.Version() || parsedVersion.IsSnapshot() {
+			t.Logf("Skipping upgrade test for version %q: same version as destination: %v snapshot: %v",
+				parsedVersion, parsedVersion.String() == define.Version(), parsedVersion.IsSnapshot())
+			continue
+		}
+
+		t.Run(fmt.Sprintf("Upgrade %s to %s", parsedVersion, define.Version()), func(t *testing.T) {
+			agentFixture, err := atesting.NewFixture(
+				t,
+				parsedVersion.String(),
+				atesting.WithFetcher(atesting.ArtifactFetcher()),
+			)
+
+			require.NoError(t, err, "error creating fixture")
+
+			err = agentFixture.Prepare(ctx)
+			require.NoError(t, err, "error preparing agent fixture")
+
+			err = agentFixture.Configure(ctx, []byte(fastWatcherCfg))
+			require.NoError(t, err, "error configuring agent fixture")
+
+			parsedUpgradeVersion, err := version.ParseVersion(define.Version())
+			require.NoErrorf(t, err, "define.Version() %q cannot be parsed as agent version", define.Version())
+			testStandaloneUpgrade(ctx, t, agentFixture, parsedVersion, parsedUpgradeVersion, "", true)
+		})
+	}
+}
+
+func TestStandaloneUpgradeToSpecificSnapshotBuild(t *testing.T) {
+	define.Require(t, define.Requirements{
+		// Stack:   &define.Stack{},
+		Local:   false, // requires Agent installation
+		Isolate: true,
```

Comment:
Why does this need to be isolated?

If it needs to remain isolated, can you add a comment on why its being isolated.

Reply:
The idea was to isolate all the standalone upgrade tests and all the fleet managed tests so that those could run in parallel on 2 separate machines